test/distributed/pipelining/test_schedule_multiproc.py

```diff
@@ -758,6 +766,194 @@ def test_v_shape_schedules(self, schedule_class, use_new_runtime):
+            }
+            assert action.sub_actions is not None
+            fwd_action = action.sub_actions[0]
+            bwd_action = action.sub_actions[1]
+
+            # Forward ========================================================
+            forward_callback(fwd_action, ctx)
+
+            # Backward ========================================================
+            backward_stage_index = bwd_action.stage_index
+            backward_stage = stage_index_to_stage[backward_stage_index]
+            backward_mb_index = bwd_action.microbatch_index
+            assert backward_mb_index is not None
+            bwd_recv_ops = schedule.bwd_recv_ops
+            is_next_stage_on_this_rank = (
+                backward_stage.stage_index + 1 in stage_index_to_stage
+            )
+            is_prev_stage_on_this_rank = (
+                backward_stage.stage_index - 1 in stage_index_to_stage
+            )
+            if (
+                not backward_stage.is_last
+                # no recv op expected for V-schedule special case (see [Note: V-schedule special case])
+                and not is_next_stage_on_this_rank
+            ):
+                if (backward_stage_index, backward_mb_index) not in bwd_recv_ops:
+                    print("HERE!")
+
+                assert (
+                    backward_stage_index,
+                    backward_mb_index,
+                ) in bwd_recv_ops, (
+                    f"Attempted to run compute {action=} before receiving input"
+                )
+                _wait_batch_p2p(
+                    bwd_recv_ops.pop((backward_stage_index, backward_mb_index))
+                )
+            loss = schedule._maybe_get_loss(backward_stage, backward_mb_index)
+            schedule.backward_counter[backward_stage_index] += 1
+            last_backward = (
+                schedule.backward_counter[backward_stage_index]
+                == schedule._n_microbatches
+            )
+            grad_scale_factor = schedule._n_microbatches if schedule.scale_grads else 1
+            backward_stage.backward_one_chunk(
+                backward_mb_index,
+                loss=loss,
+                full_backward=True,
+                last_backward=last_backward,
+            )
+            if last_backward:
+                backward_stage.scale_grads(grad_scale_factor)
+            # SEND/RECV op are avoided for special case with 2 adjacent stages on same rank
+            # see [Note: V-schedule special case]
+            if is_prev_stage_on_this_rank:
+                stage_index_to_stage[backward_stage_index - 1].set_local_bwd_input(
+                    backward_stage.get_local_bwd_output(backward_mb_index),
+                    backward_mb_index,
+                )
+
+        # Add the callback for FORWARD computation type
+
+        base_schedule.register_custom_function(FORWARD, forward_callback)
+        base_schedule.register_custom_function(OVERLAP_F_B, overlap_callback)
+
```

Comment:
I'm confused by this test. We didn't update `callback_calls` in `overlap_callback`. Do we expect `overlap_callback` to be called or not? And if yes, what's the test for `overlap_callback`?